### DISCUSSION TITLE
Install Query examples

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [microsoft/winget-cli#1415] [Q&A]
I am trying to make sense of the Install parameters. I see an option for Query and then a bunch of other parameters to customize the install. But I don't understand the distinction.

I can run this:

`winget install --query github.cli`

or this:

`winget install --id github.cli`

And get the same result. What is the purpose of the Query parameter or are there examples? I couldn't find anything in the documentation.

**Comment by denelon** [comment]
`winget install --query github.cli` is the same as `winget install github.cli`. It was named as a query early on as we were considering the possibility of wildcard support. Today it behaves more like a filter. Many of the parameters like `--id` take an argument. The first (positional) argument is treated as a positional parameter for the '--query' parameter.
